### ISSUE TITLE
部署qwen3 235b int4 gguf时，选择llama.cpp引擎发现无论如何指定gpu，它都会只使用cpu

### ISSUE THREAD

**Issue by zifeiyu-tan** [xorbitsai/inference#4336]
### System Info / 系統信息

Package                           Version
--------------------------------- ------------------
absl-py                           2.3.1
accelerate                        1.12.0
addict                            2.4.0
aiofiles                          24.1.0
aiohappyeyeballs                  2.6.1
aiohttp                           3.13.2
aioprometheus                     23.12.0
aiosignal                         1.4.0
aliyun-python-sdk-core            2.16.0
aliyun-python-sdk-kms             2.16.5
annotated-doc                     0.0.4
annotated-types                   0.7.0
anthropic                         0.75.0
antlr4-python3-runtime            4.9.3
anyascii                          0.3.3
anyio                             4.12.0
argbind                           0.3.9
argcomplete                       3.6.3
astor                             0.8.1
async-timeout                     5.0.1
attrdict                          2.0.1
attrs                             25.4.0
audioread                         3.1.0
av                                16.0.1
babel                             2.17.0
bcrypt                            5.0.0
beautifulsoup4                    4.14.3
bitsandbytes                      0.48.2
black                             26.1a1
blake3                            1.0.8
blis                              1.3.3
blobfile                          3.1.0
boto3                             1.42.5
botocore                          1.42.5
brotli                            1.2.0
cached_path                       1.8.0
cachetools                        6.2.2
catalogue                         2.0.10
cbor                              1.0.0
cbor2                             5.7.1
cdifflib                          1.2.9
certifi                           2025.11.12
cffi                              2.0.0
charset-normalizer                3.4.4
chattts                           0.2.4
click                             8.1.8
cloudpathlib                      0.23.0
cloudpickle                       3.1.2
cn2an                             0.5.23
coloredlogs                       15.0.1
compressed-tensors                0.11.0
confection                        0.1.5
conformer                         0.3.2
contourpy                         1.3.3
contractions                      0.1.73
controlnet_aux                    0.0.10
crcmod                            1.7
cryptography                      46.0.3
csvw                              3.7.0
cupy-cuda12x                      13.6.0
curated-tokenizers                0.0.9
curated-transformers              0.1.1
cycler                            0.12.1
cymem                             2.0.13
datamodel-code-generator          0.42.2
datasets                          4.4.1
dateparser                        1.1.8
decorator                         5.2.1
DeepCache                         0.1.1
Deprecated                        1.3.1
depyf                             0.19.0
diffusers                         0.36.0
dill                              0.4.0
diskcache                         5.6.3
Distance                          0.1.3
distro                            1.9.0
dlinfo                            2.0.0
dnspython                         2.8.0
docopt                            0.6.2
docstring_parser                  0.17.0
ecdsa                             0.19.1
editdistance                      0.8.1
einops                            0.8.1
einx                              0.3.0
email-validator                   2.3.0
encodec                           0.1.1
espeakng-loader                   0.2.4
eva-decord                        0.6.1
fastapi                           0.124.0
fastapi-cli                       0.0.16
fastapi-cloud-cli                 0.6.0
fastar                            0.8.0
fastrlock                         0.8.3
ffmpy                             1.0.0
filelock                          3.20.0
fire                              0.7.1
FlagEmbedding                     1.3.5
flatbuffers                       25.9.23
flatten-dict                      0.4.2
fonttools                         4.61.0
frozendict                        2.4.7
frozenlist                        1.8.0
fsspec                            2025.10.0
fugashi                           1.5.2
funasr                            1.2.7
future                            1.0.0
fvcore                            0.1.5.post20221221
g2p-en                            2.1.0
gdown                             5.2.0
genson                            1.3.0
gguf                              0.17.1
google-api-core                   2.28.1
google-auth                       2.43.0
google-cloud-core                 2.5.0
google-cloud-storage              2.19.0
google-crc32c                     1.7.1
google-resumable-media            2.8.0
googleapis-common-protos          1.72.0
gradio                            5.50.0
gradio_client                     1.14.0
groovy                            0.1.2
grpcio                            1.76.0
gruut                             2.4.0
gruut-ipa                         0.13.0
gruut_lang_de                     2.0.1
gruut_lang_en                     2.0.1
gruut_lang_es                     2.0.1
gruut_lang_fr                     2.0.2
h11                               0.16.0
hf-xet                            1.2.0
httpcore                          1.0.9
httptools                         0.7.1
httpx                             0.28.1
huggingface-hub                   0.36.0
humanfriendly                     10.0
hydra-core                        1.3.2
HyperPyYAML                       1.2.2
idna                              3.11
ijson                             3.4.0.post0
ImageIO                           2.37.2
imageio-ffmpeg                    0.6.0
importlib_metadata                8.7.0
inflect                           7.5.0
inscriptis                        2.7.0
interegular                       0.3.3
iopath                            0.1.10
ir_datasets                       0.5.11
isodate                           0.7.2
isort                             7.0.0
jaconv                            0.4.1
jamo                              0.4.1
jieba                             0.42.1
Jinja2                            3.1.6
jiter                             0.12.0
jj-pytorchvideo                   0.1.5
jmespath                          0.10.0
joblib                            1.5.2
json5                             0.12.1
jsonlines                         1.2.0
jsonschema                        4.25.1
jsonschema-specifications         2025.9.1
julius                            0.2.7
kaldifst                          1.7.17
kaldiio                           2.18.1
kiwisolver                        1.4.9
kokoro                            0.9.4
langdetect                        1.0.9
language-tags                     1.2.0
lark                              1.2.2
lazy_loader                       0.4
librosa                           0.11.0
lightning                         2.6.0
lightning-utilities               0.15.2
llama_cpp_python                  0.3.16
llguidance                        0.7.30
llvmlite                          0.44.0
lm-format-enforcer                0.11.3
loguru                            0.7.3
loralib                           0.1.2
lxml                              5.4.0
lz4                               4.4.5
Markdown                          3.10
markdown-it-py                    4.0.0
MarkupSafe                        3.0.3
matplotlib                        3.10.7
mdurl                             0.1.2
mecab-python3                     1.0.12
misaki                            0.9.4
mistral_common                    1.8.6
modelscope                        1.32.0
more-itertools                    10.8.0
mpmath                            1.3.0
msgpack                           1.1.2
msgspec                           0.20.0
multidict                         6.7.0
multiprocess                      0.70.18
munch                             4.0.0
murmurhash                        1.0.15
mypy_extensions                   1.1.0
natsort                           8.4.0
nemo_text_processing              1.1.0
networkx                          3.6.1
ninja                             1.13.0
nltk                              3.9.2
num2words                         0.5.14
numba                             0.61.2
numpy                             2.2.6
nvidia-cublas-cu12                12.8.4.1
nvidia-cuda-cupti-cu12            12.8.90
nvidia-cuda-nvrtc-cu12            12.8.93
nvidia-cuda-runtime-cu12          12.8.90
nvidia-cudnn-cu12                 9.10.2.21
nvidia-cufft-cu12                 11.3.3.83
nvidia-cufile-cu12                1.13.1.3
nvidia-curand-cu12                10.3.9.90
nvidia-cusolver-cu12              11.7.3.90
nvidia-cusparse-cu12              12.5.8.93
nvidia-cusparselt-cu12            0.7.1
nvidia-ml-py                      13.590.44
nvidia-nccl-cu12                  2.27.3
nvidia-nvjitlink-cu12             12.8.93
nvidia-nvtx-cu12                  12.8.90
omegaconf                         2.3.0
onnxruntime                       1.23.2
openai                            2.9.0
openai-harmony                    0.0.8
opencv-python-headless            4.12.0.88
optimum                           2.0.0
ordered-set                       4.1.0
orjson                            3.11.5
ormsgpack                         1.12.0
oss2                              2.19.1
outlines_core                     0.2.11
packaging                         25.0
pandas                            2.3.3
parameterized                     0.9.0
partial-json-parser               0.2.1.1.post7
pathspec                          0.12.1
peft                              0.17.1
phonemizer-fork                   3.3.2
pillow                            11.3.0
pip                               25.3
platformdirs                      4.5.1
pooch                             1.8.2
portalocker                       3.2.0
preshed                           3.0.12
proces                            0.1.7
prometheus_client                 0.23.1
prometheus-fastapi-instrumentator 7.1.0
propcache                         0.4.1
proto-plus                        1.26.1
protobuf                          6.33.2
psutil                            7.1.3
py-cpuinfo                        9.0.0
pyahocorasick                     2.2.0
pyarrow                           22.0.0
pyasn1                            0.6.1
pyasn1_modules                    0.4.2
pybase16384                       0.3.9
pybase64                          1.4.3
pycountry                         24.6.1
pycparser                         2.23
pycryptodome                      3.23.0
pycryptodomex                     3.23.0
pydantic                          2.12.3
pydantic_core                     2.41.4
pydantic-extra-types              2.10.6
pydub                             0.25.1
Pygments                          2.19.2
pykakasi                          2.3.0
pyloudnorm                        0.1.1
pynini                            2.1.6.post1
pynndescent                       0.5.13
pyparsing                         3.2.5
pypinyin                          0.55.0
pypinyin-dict                     0.9.0
PySocks                           1.7.1
python-crfsuite                   0.9.11
python-dateutil                   2.9.0.post0
python-dotenv                     1.2.1
python-jose                       3.5.0
python-json-logger                4.0.0
python-multipart                  0.0.20
pytokens                          0.3.0
pytorch-lightning                 2.6.0
pytorch-wpe                       0.0.1
pytz                              2025.2
pyworld                           0.3.5
PyYAML                            6.0.3
pyzmq                             27.1.0
quantile-python                   1.1
qwen-omni-utils                   0.0.8
qwen-vl-utils                     0.0.14
randomname                        0.2.1
ray                               2.52.1
rdflib                            7.5.0
referencing                       0.37.0
regex                             2025.11.3
requests                          2.32.5
rfc3986                           1.5.0
rich                              13.9.4
rich-toolkit                      0.17.0
rignore                           0.7.6
rpds-py                           0.30.0
rsa                               4.9.1
ruamel.yaml                       0.18.16
ruamel.yaml.clib                  0.2.15
ruff                              0.14.8
s3transfer                        0.16.0
sacremoses                        0.1.1
safehttpx                         0.1.7
safetensors                       0.7.0
scikit-image                      0.25.2
scikit-learn                      1.7.2
scipy                             1.16.3
segments                          2.3.0
semantic-version                  2.10.0
sentence-transformers             5.1.2
sentencepiece                     0.2.1
sentry-sdk                        2.47.0
setproctitle                      1.3.7
setuptools                        79.0.1
shellingham                       1.5.4
silero-vad                        6.2.0
six                               1.17.0
smart_open                        7.5.0
sniffio                           1.3.1
soundfile                         0.13.1
soupsieve                         2.8
soxr                              1.0.0
spacy                             3.8.11
spacy-curated-transformers        0.3.1
spacy-legacy                      3.0.12
spacy-loggers                     1.0.5
srsly                             2.5.2
sse-starlette                     3.0.3
starlette                         0.50.0
sympy                             1.14.0
tabulate                          0.9.0
tblib                             3.2.2
tensorboard                       2.20.0
tensorboard-data-server           0.7.2
tensorboardX                      2.6.4
termcolor                         3.2.0
textsearch                        0.0.24
thinc                             8.3.10
threadpoolctl                     3.6.0
tifffile                          2025.10.16
tiktoken                          0.12.0
timm                              1.0.22
tokenizers                        0.22.1
tomli                             2.3.0
tomlkit                           0.13.3
torch                             2.8.0
torch-complex                     0.4.4
torchaudio                        2.8.0
torchdiffeq                       0.2.5
torchmetrics                      1.8.2
torchvision                       0.23.0
tqdm                              4.67.1
transformers                      4.57.3
transformers-stream-generator     0.0.5
trec-car-tools                    2.6
triton                            3.4.0
typeguard                         4.4.4
typer                             0.20.0
typer-slim                        0.20.0
typing_extensions                 4.15.0
typing-inspection                 0.4.2
tzdata                            2025.2
tzlocal                           5.3.1
umap-learn                        0.5.9.post2
unidic-lite                       1.0.8
unlzw3                            0.2.3
uritemplate                       4.2.0
urllib3                           2.6.1
uv                                0.9.16
uvicorn                           0.38.0
uvloop                            0.22.1
vector-quantize-pytorch           1.17.3
verovio                           5.7.0
vllm                              0.11.0
vocos                             0.1.0
warc3-wet                         0.2.5
warc3-wet-clueweb09               0.2.5
wasabi                            1.1.3
watchfiles                        1.1.1
weasel                            0.4.3
websockets                        15.0.1
Werkzeug                          3.1.4
wetext                            0.1.2
wget                              3.2
wheel                             0.45.1
wrapt                             2.0.1
x-transformers                    2.11.23
xformers                          0.0.32.post1
xgrammar                          0.1.25
xinference                        1.14.0
xllamacpp                         0.2.6
xoscar                            0.8.0
xxhash                            3.6.0
yacs                              0.1.8
yarl                              1.22.0
zipp                              3.23.0
zlib-state                        0.1.10

### Running Xinference with Docker? / 是否使用 Docker 运行 Xinfernece？

- [ ] docker / docker
- [x] pip install / 通过 pip install 安装
- [ ] installation from source / 从源码安装

### Version info / 版本信息

xinference                        1.14.0

### The command used to start Xinference / 用以启动 xinference 的命令

xinference launch --model-name qwen3 --model-type LLM --model-engine llama.cpp --model-format ggufv2 --size-in-billions 235 --quantization Q4_1 --model-uid qwen3_235b_int4 --gpu-idx 1,3 --model-path /home/user/.cache/modelscope/hub/unsloth/Qwen3-235B-A22B-GGUF/Q4_1/Qwen3-235B-A22B-Q4_1-merged.gguf --n-gpu auto --replica 1 --enable_thinking true --reasoning_content true
或者
xinference launch --model-name qwen3 --model-type LLM --model-engine llama.cpp --model-format ggufv2 --size-in-billions 235 --quantization Q4_1 --model-uid qwen3_235b_int4 --model-path /home/user/.cache/modelscope/hub/unsloth/Qwen3-235B-A22B-GGUF/Q4_1/Qwen3-235B-A22B-Q4_1-merged.gguf --n-gpu auto --replica 1 --enable_thinking true --reasoning_content true

### Reproduction / 复现过程

激活环境
启动xinference-local -H 0.0.0.0 -p 9997
然后用上述命令启动模型

### Expected behavior / 期待表现

能够使用gpu完成部署

**Comment by OliverBryant** [comment]
因为xinference默认安装的xllamacpp是cpu版本，可以按照如下操作安装gpu版xllamacpp：
1. pip uninstall -y xllamacpp && pip install xllamacpp -i https://xorbitsai.github.io/xllamacpp/whl/cu128
2. 如果安装后启动发现llama cpp引擎不可用，报错：：Failed to import xllamacpp: lib/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by miniconda3/envs/xinf/lib/
  python3.10/site-packages/xllamacpp/xllamacpp.cpython-310-x86_64-linux-gnu.so)，可以执行如下命令：conda install -c conda-forge "libstdcxx-ng>=13" "libgcc-ng>=13"，将 libstdc++ 升级到包含 GLIBCXX_3.4.30，可以解决，或者安装cu124等较低版本
